src/Request.php

```diff
@@ -91,6 +92,7 @@ public function getMethod()
 
     public function withMethod($method)
     {
+        $this->assertMethod($method);
         $new = clone $this;
         $new->method = strtoupper($method);
```

Comment:
According to the tests the strtoupper should not be here.. the current PR targets 1,x.. Can this be fixed on v1. or it should be fixed on version 2.x?